sporkfish/lichess_bot/lichess_bot_berserk.py

```diff
@@ -144,13 +145,39 @@ def set_pos_and_play_move(
                 num_moves = len(state["moves"].split())
                 prev_moves = state["moves"]
                 set_pos_and_play_move(num_moves, color, prev_moves, game_id, state)
+            elif state["type"] == "opponentGone" and state["gone"]:
+                time.sleep(8)
+                self.client.board.claim_victory(game_id)
+
+    def should_accept_challenge(self, event):
+        bool_challenge = True
+        if (
+            event["challenge"]["speed"] in {"rapid", "bullet", "blitz"}
+            and event["challenge"]["variant"]["key"] == "standard"
+        ):
+            bool_challenge
+        else:
+            bool_challenge = False
+        return bool_challenge
 
     def run(self) -> None:
         """
         Start the Lichess bot, listening to incoming events sequentially and playing games accordingly.
         """
-
         events = self.client.bots.stream_incoming_events()
         for event in events:
-            if event.get("type") == "gameStart":
+            if event.get("type") == "challenge":
```

Comment:
This is getting a bit long. Shall we just have a `_handle_game_states` function as you suggested in your previous comments?

Reply:
Alternatively you could have a bunch of "handlers" (i.e. functions) for each state and store these in a dictionary, corresponding to the event "type". Then just dispatch using event.get("type"). IMO this is cleanest but I can do it myself if unclear.